### ISSUE TITLE
Option to update a plugin to a specific version

### ISSUE THREAD

**Issue by acchen97** [elastic/logstash#4369]
There are use cases where a user may need to update a plugin to a particular version that isn't the latest.  There should be a --version flag for "bin/plugin update" to enable this.

Related to: https://github.com/elastic/logstash/issues/3334

**Comment by purbon** [comment]
In my opinion the most important thing to provide here is idempotent updates command, and this will be hard to archive only with a ```---version``` flag, obviously this should be be skip as is a good feature to have for the update command. I some challenges here:

* We provide no feedback to our users when hitting plugin dependencies conflicts. I started an small effort, that needs a bit of polishing at (https://github.com/elastic/logstash/pull/4288) to address this problem.
* We should provide idempotent updates across machines, this will be challenging only with adding the ``---version``` flag, due to possible inter plugin conflicts. Some of this might be archived by sharing the install status (aka ```Gemfile.lock file```).


what do you think?

**Comment by acchen97** [comment]
@purbon I agree idempotent updates are important and I think the --version flag should solve it if we resolve the conflicts within the update operation (update or uninstall dependent plugins).

The current way to install a specific version of a plugin is to uninstall all plugins that depend on it first, then uninstall the actual plugin and reinstall it with "install --version".

Uninstall command user feedback is another item, success and failure (+reasons why) should be clearly displayed and I'm +1 on more feedback here.

**Comment by ppuschmann** [comment]
Let's give an example to this request:

Scenario: 
* Kafka 0.9 Cluster shall get used with Logstash 2.4

According to https://github.com/logstash-plugins/logstash-output-kafka we should use plugin version 4.0.0 to use the correct plugin version to work with Kafka 0.9 brokers.

-> How do I install this version 4.0.0 on the cli?

`bin/logstash-plugin install logstash-output-kafka --version 4.0.0` does not work.

Right now I somehow have to do this manually, wasting some precious time on how to automate/script this.

**Comment by raiusa** [comment]
@ppuschmann I have the same use case and trying to make it working. Can you please guide me what manual step you have done?

**Comment by acchen97** [comment]
@wongchao the current workaround here is to uninstall and reinstall with the desired plugin version.

**Comment by ppuschmann** [comment]
@raiusa I have no workaround. 
We "fixed" this by installing a newer version of Logstash that already shipped with a plugin that is (more) compatible to our kafka-broker version.
